web/src/main/java/org/springframework/security/web/authentication/ott/DefaultGenerateOneTimeTokenRequestResolver.java

```diff
@@ -37,13 +39,15 @@ public final class DefaultGenerateOneTimeTokenRequestResolver implements Generat
 
 	private Duration expiresIn = DEFAULT_EXPIRES_IN;
 
+	private Supplier<String> tokenValueFactory = () -> UUID.randomUUID().toString();
+
 	@Override
 	public GenerateOneTimeTokenRequest resolve(HttpServletRequest request) {
 		String username = request.getParameter("username");
 		if (!StringUtils.hasText(username)) {
 			return null;
 		}
-		return new GenerateOneTimeTokenRequest(username, this.expiresIn);
+		return new GenerateOneTimeTokenRequest(username, this.expiresIn, this.tokenValueFactory.get());
```

Comment:
I think we should not generate token value here, since it breaks the semantic of `GenerateOneTimeTokenRequestResolver` which is:
"A strategy for resolving a `GenerateOneTimeTokenRequest` from the `HttpServletRequest`".

It might also be confusing for users who provide a custom `tokenValueFactory` and notice that their factory gets called before `OneTimeTokenService#generate` method call.

So I would consider passing `Supplier<String> tokenValueFactory` to `GenerateOneTimeTokenRequest` and call it in the `OneTimeTokenService#generate` method.